// @rhinestone/module-sdk/_cjs/module/zk-email-recovery/universal-email-recovery/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getModuleAddress = exports.CANCEL_EXPIRED_RECOVERY_COOLDOWN = exports.MINIMUM_RECOVERY_WINDOW = exports.MAX_NUMBER_OF_GUARDIANS = exports.MAX_VALIDATORS = exports.UNIVERSAL_EMAIL_RECOVERY_ADDRESS__ETH_SEPOLIA = exports.UNIVERSAL_EMAIL_RECOVERY_ADDRESS__BASE = void 0;
const chains_1 = require("viem/chains");
exports.UNIVERSAL_EMAIL_RECOVERY_ADDRESS__BASE = '0x36A470159F8170ad262B9518095a9FeD0824e7dD';
exports.UNIVERSAL_EMAIL_RECOVERY_ADDRESS__ETH_SEPOLIA = '0x8ECcb707C4770239D7e95743cd01aaA72d6D313E';
exports.MAX_VALIDATORS = 32;
exports.MAX_NUMBER_OF_GUARDIANS = 32;
exports.MINIMUM_RECOVERY_WINDOW = 2 * 24 * 60 * 60;
exports.CANCEL_EXPIRED_RECOVERY_COOLDOWN = 24 * 60 * 60;
const getModuleAddress = (chainId) => {
    if (chainId === chains_1.sepolia.id) {
        return exports.UNIVERSAL_EMAIL_RECOVERY_ADDRESS__ETH_SEPOLIA;
    }
    if (chainId === chains_1.base.id) {
        return exports.UNIVERSAL_EMAIL_RECOVERY_ADDRESS__BASE;
    }
    throw new Error('Unsupported chain');
};
exports.getModuleAddress = getModuleAddress;
